lib/onebox/engine/pdf_onebox.rb

```diff
@@ -0,0 +1,37 @@
+require 'pdf-reader'
+
+module Onebox
+  module Engine
+    class PdfOnebox
+      include Engine
+      include LayoutSupport
+
+      matches_regexp(/^(https?:)?\/\/.*\.pdf(\?.*)?$/i)
+      always_https
+
+      private
+
+      def data
+        html_entities = HTMLEntities.new
+        pdf_response = get_pdf_response
+        pdf_data = pdf_response.info
+
+        result = { link: link }
+        result['title'] = unless pdf_data[:Title].blank?
+          html_entities.decode(Onebox::Helpers.truncate(pdf_data[:Title].force_encoding("UTF-8").scrub.strip, 80))
+        else
+          "PDF File"
+        end
+        result['description'] = html_entities.decode(Onebox::Helpers.truncate(pdf_data[:Subject].force_encoding("UTF-8").scrub.strip, 250)) rescue nil
+        result['author'] = pdf_data[:Author] unless pdf_data[:Author].blank?
+        result
+      end
+
+      def get_pdf_response
+        PDF::Reader.new(open(@url, read_timeout: Onebox.options.timeout))
```

Comment:
Looking at limited [documentation](https://github.com/yob/pdf-reader#usage) it loads the whole file in memory.

I looked at two other [alternatives](https://www.ruby-toolbox.com/categories/PDF_Processing) but they do not support IO stream and the file needs to be downloaded on disk to fetch information/metadata.